src/pymmcore_gui/widgets/image_preview/_ndv_preview.py

```diff
@@ -66,9 +66,18 @@ def _setup_viewer(self) -> None:
             return  # pragma: no cover
 
         self._core_dtype = core_dtype
+        # START OF CHANGE
+        # Coerce non-standard integer dtypes to a supported numpy dtype for the buffer.
+        # A 12-bit image is stored in a 16-bit buffer.
+        dtype_str, shape = core_dtype
+        if dtype_str == "uint12":
+            buffer_dtype = ("uint16", shape)
+        else:
+            buffer_dtype = core_dtype
+
         self._viewer.data = self._buffer = RingBuffer(
-            max_capacity=100, dtype=core_dtype
-        )
+        max_capacity=100, dtype=buffer_dtype)
+        # END OF CHANGE
```

Comment:
i think it's probably safe to do this higher up in `_get_core_dtype_shape`... at line 60:
```python
return (f"uint{bits}", shape)
```